### ISSUE TITLE
Threshold for paralleization

### ISSUE THREAD

**Issue by scheherzade** [numpy/numpy#12311]
I'm running my code with diffrenet number of cores but I only see it scaling when matrix size is larger than a value. I was wondering if there is a threshold in NumPy for different operations so that it would run in parallel? And if so where can I find them?

**Comment by seberg** [comment]
Yes, it is normally 500. You can search the source code for `NPY_BEGIN_THREADS_THRESHOLDED` and `NPY_BEGIN_THREADS`. Releasing and grabbing the GIL is not super fast, so a threshold makes normally sense.

**Comment by seberg** [comment]
This is of course only if you do threading yourself. If you do not, another cause is the BLAS you are using, which may have similar rules.
